browser/tor/tor_profile_service_impl.cc

```diff
@@ -173,8 +175,10 @@ void TorProfileServiceImpl::SetNewTorCircuit(WebContents* tab) {
   }
   auto proxy_lookup_client =
       TorProxyLookupClient::CreateTorProxyLookupClient(std::move(callback));
+  url::Origin origin = url::Origin::Create(url);
```

Comment:
this has larger implications cc @darkdh @riastradh-brave

Reply:
We get tor proxy config result directly in `ProxyResolutionService::TryToCompleteSynchronously` and  create username in `ProxyConfigServiceTor::CircuitIsolationKey`. so we doesn't use `net::NetworkIsolationKey` even the interface requires it. Or do you suggest we should consider `net::NetworkIsolationKey` when calculating our circuit isolation key?

Reply:
well, what I'm getting at is that this could simplify it and we probably should be using `network_isolation_key`. Not saying anything should happen in this PR, but maybe we need a follow-up issue to look at this?

Reply:
follow-up issue created https://github.com/brave/brave-browser/issues/7955